src/System.Device.Gpio.Tests/LibGpiodDriverTests.cs

```diff
@@ -71,5 +71,37 @@ static void PinChanged(object sender, PinValueChangedEventArgs args)
                 gc.UnregisterCallbackForPinValueChangedEvent(InputPin, PinChanged);
             }
         }
+
+        /// <summary>
+        /// Ensure leaking instances of the driver doesn't cause a segfault
+        /// See #1849 for a description of this test case
+        /// </summary>
+        [Fact]
+        public void LeakingDriverDoesNotCrash()
```

Comment:
also looking at implementation seems SetPinMode is calling ReleaseLock. Can you also add a test where we change mode several times and see if anything breaks?

Reply:
That test exists and was added when the current implementation was done. Without that call to ReleaseLock, SetPinMode fails when done repeatedly.